lib/api/controllers/funnelController.js

```diff
@@ -127,7 +131,10 @@ class FunnelController {
 
     // resolves the callback immediately if a slot is available
     if (this.concurrentRequests < this.kuzzle.config.limits.concurrentRequests) {
-      return callback(null);
+      if (this.cachedIds[request.id]) {
+        delete this.cachedIds[request.id];
```

Comment:
As seen together, this particular situation cannot occur be we still may have had a problem where a request could potentially be dropped from the cache queue before it is actually processed.

Should be fix by last commit